x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -38,11 +38,47 @@ export interface IngestPipelineParams {
   run_ml_inference: boolean;
 }
 
+export interface FilteringRule {
+  created_at: string;
+  field: string;
+  id: string;
+  order: number;
+  policy: string;
+  rule: string;
+  updated_at: string;
+  value: string;
+}
+
+export interface FilteringValidation {
+  ids: string[];
+  messages: string[];
+}
+
+export interface FilteringRules {
+  advanced_snippet: {
+    created_at: string;
+    updated_at: string;
+    value: object;
+  };
+  rules: FilteringRule[];
+  validation: {
+    errors: FilteringValidation[];
+    state: string;
+  };
+}
+
+export interface FilteringConfig {
+  active: FilteringRules;
+  domain: string;
+  draft: FilteringRules;
+}
+
 export interface Connector {
   api_key_id: string | null;
   configuration: ConnectorConfiguration;
   description: string | null;
   error: string | null;
+  filtering: FilteringConfig[] | null;
```

Comment:
question: Do we want to make this a mandatory property and initialize it as an empty array instead?

Reply:
I was thinking this would be a more friendly way of not breaking all <=8.5 deployments, without requiring us to write a migration to add this. WDYT?